hybridse/src/udf/default_udf_library.cc

```diff
@@ -1612,6 +1612,21 @@ void DefaultUdfLibrary::InitTimeAndDateUdf() {
             @since 0.1.0
         )");
 
+    RegisterExternal("dayofyear")
+        .args<int64_t>(static_cast<int32_t (*)(int64_t)>(v1::dayofyear))
+        .args<Timestamp>(static_cast<int32_t (*)(Timestamp*)>(v1::dayofyear))
+        .args<Date>(static_cast<int32_t (*)(Date*)>(v1::dayofyear))
+        .doc(R"(
+            @brief Return the day of year for a timestamp or date.
+
+            Example:
+            @code{.sql}
+                select dayofyear(timestamp(1590115420000));
+                -- output 143
+            @endcode
+            @since 0.1.0
+        )");
```

Comment:
@since 0.4.0.
> `@since`
> command to specify the production version when the function was added to OpenMLDB. The version can be obtained from the project's CMakeList.txt: ${OPENMLDB_VERSION_MAJOR}.${OPENMLDB_VERSION_MINOR}.${OPENMLDB_VERSION_BUG}